internal/manager/manager.go

```diff
@@ -175,21 +181,37 @@ func (m *manager) connectRegistry() (err error) {
 
 func (m *manager) startServers(nsmMgr *registry.NetworkServiceEndpoint, server *grpc.Server) {
 	var wg sync.WaitGroup
-	for _, u := range m.configuration.ListenOn {
-		listenURL := *u
+	for i := 0; i < len(m.configuration.ListenOn); i++ {
+		listenURL := &m.configuration.ListenOn[i]
 		wg.Add(1)
 
 		go func() {
+			if listenURL.Scheme == unixSchema || listenURL.Scheme == "" {
+				basePath := path.Dir(listenURL.Path)
+				_, statRes := os.Stat(basePath)
+				if os.IsNotExist(statRes) {
+					logrus.Infof("Unix socket folder %v not exists, Trying to create", basePath)
+					if err := os.MkdirAll(basePath, os.ModePerm); err != nil {
+						logrus.Fatalf("Failed to %v. Could not serve %v", err, listenURL.String())
+						wg.Done()
+						return
+					}
+				}
+				if listenURL.Scheme == "" {
+					// Replace schema to be unix if not specified.
+					listenURL.Scheme = unixSchema
+				}
+			}
```

Comment:
Can we move this code into `grpcutils.ListenAndServe`?

Reply:
Also, I think we need to check is the socket file exists and if yes then we should remove it.

Reply:
Moved this function and added a fix for existing socket files in SDK https://github.com/networkservicemesh/sdk/pull/440